src/compiler/checker.ts

```diff
@@ -22881,9 +22923,15 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         return undefined;
     }
 
-    function isTupleTypeStructureMatching(t1: TupleTypeReference, t2: TupleTypeReference) {
+    function isTupleTypeStructureMatching(t1: TupleTypeReference, t2: TupleTypeReference, strictVariableElementsComparison: boolean) {

```

Comment:
I'm not super fond of this change. It feels a little bit icky that we end up with different "modes" for comparing tuple type structures.

However, the name of the function is almost perfect for my use case - I couldn't figure out a better name for a "forked" implementation of this one. There is only one existing call to this helper function and it can be found here:
https://github.dev/microsoft/TypeScript/blob/4ba756fdc847115bc1ff5032ed06e9bed60f16b2/src/compiler/checker.ts#L24408-L24415

Note that this whole function was already a little bit specific to this one call site - since that one call site **allows** a required element to match an optional one. So, in a sense, I do a similar thing by allowing a variadic one to match a rest one. 

I'm also debating with myself - if in my case, I shouldn't disallow required elements from matching optional ones. I need to think about this more though, perhaps I will find a different solution for the cases related to the optional elements that I have to fix. If it comes to that - then I will probably refactor this new parameter to accept a bitmask.